src/main/java/org/snomed/snowstorm/core/data/domain/Annotation.java

```diff
@@ -0,0 +1,165 @@
+	private String annotationValue;
+	@JsonView(value = View.Component.class)
+	private String annotationLanguage;
+
+	@Override
+	public String getAnnotationId() {
+		return annotationId;
+	}
+
+	public void setAnnotationId(String annotationId) {
+		this.annotationId = annotationId;
+	}
+
+	@Override
+	public String getAnnotationTypeId() {
+		return annotationTypeId;
+	}
+
+	public void setAnnotationTypeId(String annotationTypeId) {
+		this.annotationTypeId = annotationTypeId;
+	}
+
+	@JsonIgnore
+	public ConceptMini getAnnotationType() {
+		return annotationType;
+	}
+
+	@Override
+	@JsonView(value = View.Hidden.class)
+	public String getMemberId() {
+		return super.getMemberId();
+	}
+
+	@Override
+	@JsonView(value = View.Hidden.class)
+	public Map<String, String> getAdditionalFields() {
+		return super.getAdditionalFields();
+	}
+
+	@Override
+	@JsonView(value = View.Component.class)
+	public TermLangPojo getAnnotationTypePt() {
+		return annotationType != null ? annotationType.getPt() : null;
+	}
+
+	public void setAnnotationType(ConceptMini annotationType) {
+		this.annotationType = annotationType;
+	}
+
+	@Override
+	public String getAnnotationValue() {
+		return annotationValue;
+	}
+
+	public void setAnnotationValue(String annotationValue) {
+		this.annotationValue = annotationValue;
+	}
+
+	@Override
+	public String getAnnotationLanguage() {
+		return annotationLanguage;
+	}
+
+	public void setAnnotationLanguage(String annotationLanguage) {
+		this.annotationLanguage = annotationLanguage;
+	}
+
+	public Annotation fromRefsetMember(ReferenceSetMember fromMember) {
+		setAnnotationId(fromMember.getMemberId());
+		setMemberId(fromMember.getMemberId());
+		setRefsetId(fromMember.getRefsetId());
+		setModuleId(fromMember.getModuleId());
+		setActive(fromMember.isActive());
+		setEffectiveTimeI(fromMember.getEffectiveTimeI());
+		setReferencedComponentId(fromMember.getReferencedComponentId());
+		setReleased(fromMember.isReleased());
+		setAnnotationTypeId(fromMember.getAdditionalField(AnnotationFields.ANNOTATION_TYPE_ID));
+
+		String annotationValue = fromMember.getAdditionalField(AnnotationFields.ANNOTATION_VALUE);
+		if (StringUtils.isNotEmpty(annotationValue) && ANNOTATION_LANGUAGE_TYPE_PATTERN.matcher(annotationValue).matches()) {
+			setAnnotationLanguage(annotationValue.substring(1, annotationValue.indexOf("\"")).trim());
+			setAnnotationValue(annotationValue.substring(annotationValue.indexOf("\"") + 1, annotationValue.lastIndexOf("\"")));
+		} else {
+			setAnnotationValue(annotationValue);
+		}
+		return this;
+	}
+
+	public ReferenceSetMember toRefsetMember() {
+		String refsetId = getRefsetId() != null ? getRefsetId() : Concepts.ANNOTATION_REFERENCE_SET;
+		String annotationId = getAnnotationId() != null ? getAnnotationId() : UUID.randomUUID().toString();
+		ReferenceSetMember member = new ReferenceSetMember(annotationId, getEffectiveTimeI(), isActive(), getModuleId(), refsetId, getReferencedComponentId());
```

Comment:
What's the expected payload when creating an Annotation @QuyenLy87? When creating a Concept without a module id, the default branch module is applied. When creating an Annotation without a module id, none is specified so null reaches the database.

Reply:
Hi @dmcgihtsdo, Are you sure that you provided the default module for the branch? If yes, it will be used for Annotation while saving the concept.

Reply:
Good spot @QuyenLy87, yes I hadn't supplied the default module. However, the default module isn't used for International, only Extensions.

Reply:
Here is how we populate the default module ID if not specified for RefsetMember
https://github.com/IHTSDO/snowstorm/blob/develop/src/main/java/org/snomed/snowstorm/core/data/services/ConceptUpdateHelper.java#L290

Reply:
Ah,, Yes, I see what you mean. I updated the Annotation class. If no module is provided, then use CORE_MODULE instead.